serve.go

```diff
@@ -24,19 +27,46 @@ func (s *Server) Run() error {
 // It also generates the OpenAPI spec and outputs it to a file, the UI, and a handler (if enabled).
 func (s *Server) RunTLS(certFile, keyFile string) error {
 	s.isTLS = true
-
-	s.setup()
-	return s.Server.ListenAndServeTLS(certFile, keyFile)
+	if err := s.setup(); err != nil {
+		return err
+	}
+	return s.Server.ServeTLS(s.listener, certFile, keyFile)
 }
 
-func (s *Server) setup() {
+func (s *Server) setup() error {
+	if err := s.setupDefaultListener(); err != nil {
+		return err
+	}
 	go s.OutputOpenAPISpec()
 	s.printStartupMessage()
 
 	s.Server.Handler = s.Mux
 	if s.corsMiddleware != nil {
 		s.Server.Handler = s.corsMiddleware(s.Server.Handler)
 	}
+	return nil
+}
+
+// setupDefaultListener creates a default (non-TLS) listener if none is already configured.
+// If a listener is already set, this method does nothing.
+// Returns an error if the listener cannot be created (e.g., address binding issues).
+func (s *Server) setupDefaultListener() error {
+	if s.listener != nil {
+		return nil // Listener already exists, no action needed.
+	}
+	addr := s.Server.Addr
+	if addr == "" {
```

Comment:
This check isn't really needed. We always default when it is not provided. Also this would only bind the loopback address which would make it confusing for users on why they can't reach the service. The .Listen below should should just take whatever `s.Server.Addr` is.